tasks/run-file-updates/tests/mocks.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+set -eux
+
+# mocks to be injected into task step scripts
+
+function internal-request() {
+  TIMEOUT=30
+  END_TIME=$(date -ud "$TIMEOUT seconds" +%s)
+
+  echo Mock internal-request called with: $*
+  echo $* >> $(workspaces.data.path)/mock_internal-request.txt
+
+  # since we put the IR in the background, we need to be able to locate it so we can
+  # get the name to patch it. We do this by tacking on another random label that we can use
+  # to select with later.
+  rando=$(openssl rand -hex 12)
+  /home/utils/internal-request "$@" -l "internal-services.appstudio.openshift.io/test-id=$rando" &
+
+  sleep 2
+  NAME=
+  while [[ -z ${NAME} ]]; do
+    if [ "$(date +%s)" -gt "$END_TIME" ]; then
+        echo "ERROR: Timeout while waiting to locate InternalRequest"
+        echo "Internal requests:"
+        kubectl get internalrequest --no-headers -o custom-columns=":metadata.name" \
+            --sort-by=.metadata.creationTimestamp
+        exit 124
+    fi
+
+    NAME=$(kubectl get internalrequest -l "internal-services.appstudio.openshift.io/test-id=$rando" \
+        --no-headers -o custom-columns=":metadata.name" \
+        --sort-by=.metadata.creationTimestamp | tail -1)
+    if [ -z $NAME ]; then
+        echo "Warning: Unable to get IR name"
+        sleep 2
+    fi
+  done
+  echo "IR Name: $NAME"
+  if [[ "$*" == *"ref=fail2"* ]]; then
+      set_ir_status $NAME Failure 2
+  elif [[ "$*" == *"ref=fail"* ]]; then
+      set_ir_status $NAME Failure 5
+  else
+      set_ir_status $NAME Succeeded 5
+  fi
+  wait -n
+}
+
+function set_ir_status() {
+    NAME=$1
+    REASON=$2
+    DELAY=$3
+    echo Setting status of $NAME to reason $REASON in $DELAY seconds... >&2
+    sleep $DELAY
+    PATCH_FILE=$(workspaces.data.path)/${NAME}-patch.json
+    status="Succeeded"
+    mockPipelineRunName="$(openssl rand -hex 12)"
+    mockTaskRunName="${mockPipelineRunName}-tr"
+    if [ "${REASON}" == "Failure" ]; then
+      status="Failed"
+      cat > $PATCH_FILE << EOF
+{
+  "status": {
+    "results": {
+      "mockPipelineRunName": "${mockPipelineRunName}",
```

Comment:
Shouldn't this be `internalRequestPipelineRunName`? Same for the task. I am confused as to how your test passed because you test for this

Reply:
nice catch!

```
[check-result : check-result] + internalRequestPipelineRunName=null
[check-result : check-result] ++ jq -r --argjson i 2 '.items[$i].status.results.internalRequestTaskRunName'
[check-result : check-result] + internalRequestTaskRunName=null
[check-result : check-result] + test null '!=' ''
[check-result : check-result] + test null '!=' ''
```

fixed